libr/core/cmd_anal.c

```diff
@@ -9703,6 +9703,20 @@ R_API void cmd_agfb2(RCore *core, const char *s) {
 	free (pix);
 }
 
+static inline bool fcn_siwtch_mermaid(RAnalBlock *b, RStrBuf *buf) {
+	if (b->switch_op) {
+		r_return_val_if_fail (b->switch_op->cases, false);
+		RListIter *itt;
+		RAnalCaseOp *c;
+		r_list_foreach (b->switch_op->cases, itt, c) {
+			if (!r_strbuf_appendf (buf, "\t_0x%" PFMT64x " --> _0x%" PFMT64x ": Case %" PFMT64d "\n", b->addr, c->addr, c->value)) {
```

Comment:
Do this format works with spaces or tabs are mandatory? Better use spaces for r2 printers as long as tabs mess with visual/panels

Reply:
Is c->value ut64? Better add a cast or use %d

Reply:
yeah c->value is a ut64 for some reason. I cast it to int so I could just use %d. There shouldn't be more then 0x7fffffff cases and it should be possible to have negative cases.

That said, I can fix it any way you like...

Reply:
Good catch